### ISSUE TITLE
Option to build vignettes on install

### ISSUE THREAD

**Issue by jistria** [rstudio/renv#585]
Is it currently possible to ask `renv` to build vignettes when it installs a package from source? 

If not, could you please consider adding this as an option? Perhaps something similar to the `build_vignettes` parameter of `devtools::install`. 

Thanks for your time!

**Comment by kevinushey** [comment]
Normally, vignettes are built at package build time, as opposed to package install time. Are you referring explicitly to the case where one want to install a package from sources; e.g.

```
renv::install("~/path/to/package/folder")
```

or do you mean specifically for already-built source package tarballs?

**Comment by jistria** [comment]
> Are you referring explicitly to the case where one want to install a package from sources

Yes. In my specific case, I am asking my colleagues to install a package whose source code is hosted on our company's internal GitLab server.

**Comment by MajoroMask** [comment]
Second this function request!

For now our work-around is :

- `git clone remote dev` to get source to local
- `devtools::build(binary = TRUE)` to build the package into  binary status
-  `renv::install("path/to/binary.tgz")`(our team work on MacOS) to test our package

Could be more straightforward if we just have this `renv::install("GitLab/repos", build_vignettes = TRUE)`

Thanks a lot!

**Comment by kevinushey** [comment]
This has now been implemented in the development version of `renv`. Sorry it took us so long to get around to it. If you have time, can you test and report back?

**Comment by jistria** [comment]
Thanks very much for your time! I appreciate that this is a low priority request.

I've tested with the current version of `renv` (`0.15.2.7`), and when installing a package from source, `renv::install` builds vignettes if the packages required to build the vignettes have previously been installed (i.e. packages listed under the Suggests heading in the DESCRIPTION file; `knitr`, `rmarkdown`, any other packages referenced by the vignettes). 

Would you consider having `renv::install` install the packages listed under the Suggests heading when building vignettes? Apologies if there is already some way of doing this that I haven't realised.

**Comment by kevinushey** [comment]
> Would you consider having renv::install install the packages listed under the Suggests heading when building vignettes?

This seems necessary if we want to be able to build package vignettes during package install; I'll see if we can explore this soon.

**Comment by hadley** [comment]
Now tracked in #1201
